README.md

```diff
@@ -44,6 +44,299 @@ This is an experimental framework that aims to combine the lessons learned from
+This requires to define the annotation container as follows:
+
+```python
+class MyDocumentWithAlignment(Document):
+    text_a: str
+    text_b: str
+    # `named_targets` defines the mapping from `TARGET_NAMES` to data fields
+    my_alignments: AnnotationList[Alignment] = annotation_field(named_targets={"text1": "text_a", "text2": "text_b"})
+```
+
+Note that `text1` and `text2` can also target the same field.
+
+</details>
+<details>
+<summary>
+
+##### (De-)Serialization of Annotations
+
+</summary>
+
+As usual for dataclasses, annotations can be converted to json like objects with `.asdict()`. However, they can be
+also created with `MyAnnotation.fromdict(dct, annotation_store)`. Both methods are required because documents and
+their annotations are created on the fly when working with PIE datasets (see below).
+
+Sometimes, it is required to overwrite both methods. This is the case when targeting another annotation field. Consider
+the following example where `head` and `tail` are entries from another annotation field:
+
+```python
+@dataclass(eq=True, frozen=True)
+class BinaryRelation(Annotation):
+    head: Span
+    tail: Span
+    label: str
+
+    def asdict(self) -> Dict[str, Any]:
+        # Convert the annotations to their ids.
+        # We use the _asdicts() method with overrides to avoid converting the original
+        # entries to dicts in the first place (this can slow down the preprocessing a lot).
+        dct = self._asdict(overrides={"head": self.head._id, "tail": self.tail._id})
+        return dct
+
+    @classmethod
+    def fromdict(
+        cls,
+        dct: Dict[str, Any],
+        annotation_store: Optional[Dict[int, Annotation]] = None,
+    ):
+        # copy to not modify the input
+        tmp_dct = dict(dct)
+        # get the annotations by their ids
+        tmp_dct["head"] = resolve_annotation(tmp_dct["head"], store=annotation_store)
+        tmp_dct["tail"] = resolve_annotation(tmp_dct["tail"], store=annotation_store)
+        return super().fromdict(tmp_dct, annotation_store)
+```
+
+Here it is necessary to replace the referenced `Span` annotations with their ids during serialization because
+we save them already in the respective annotation field. Thus, we also have to replace the ids with the actual
+annotations during construction. This can be easily done with the helper method
+`resolve_annotation(id_or_annotation, store)`.
+
+</details>
+</details>
+<details>
+<summary>
+
+### 🔤 ⇔ 🔢 Taskmodule
+
+</summary>
+
+The taskmodule is responsible for converting documents to model inputs and back. For that purpose, it requires the
+user to implement the following methods:
+
+-   `encode_input`: Taking one document, create one or multiple `TaskEncoding`s. A `TaskEncoding` represents an
+    example that will be passed to the model later on. It is a container holding `inputs`, optional `targets`, the
+    original `document`, and `metadata`. Note that `encode_input` should not assign a value to `targets`.
```

Comment:
so "targets" here has the meaning of e.g. 'classification label' ? (as opposed to above for annotations)

Reply:
good point. then renaming the annotation related targets to `references`? or what do you think would a good name that reflects the concept?